src/testing/time_refinement_study_advection.cpp

```diff
@@ -0,0 +1,135 @@
+#include "time_refinement_study_advection.h"
+#include "flow_solver.h"
+#include "flow_solver_cases/periodic_1D_flow.h"
+#include "cmath"
+
+namespace PHiLiP {
+namespace Tests {
+
+template <int dim, int nstate>
+TimeRefinementStudyAdvection<dim, nstate>::TimeRefinementStudyAdvection(
+        const PHiLiP::Parameters::AllParameters *const parameters_input,
+        const dealii::ParameterHandler &parameter_handler_input)  
+        : TestsBase::TestsBase(parameters_input),
+         parameter_handler(parameter_handler_input)
+{}
+
+
+
+template <int dim, int nstate>
+ Parameters::AllParameters TimeRefinementStudyAdvection<dim, nstate>::reinit_params_and_refine_timestep(int refinement) const{
+     dealii::ParameterHandler parameter_handler;
+     PHiLiP::Parameters::AllParameters::declare_parameters (parameter_handler);
+     PHiLiP::Parameters::AllParameters parameters;
+     parameters.parse_parameters(parameter_handler);
+
+
+     //parameters = this->all_parameters; //This doesn't work, operator = not defined. Block below copies params one at a time.
+
+     // Copy all parameters
+     parameters.manufactured_convergence_study_param = this->all_parameters->manufactured_convergence_study_param;
+     parameters.ode_solver_param = this->all_parameters->ode_solver_param;
+     parameters.linear_solver_param = this->all_parameters->linear_solver_param;
+     parameters.euler_param = this->all_parameters->euler_param;
+     parameters.navier_stokes_param = this->all_parameters->navier_stokes_param;
+     parameters.reduced_order_param= this->all_parameters->reduced_order_param;
+     parameters.burgers_param = this->all_parameters->burgers_param;
+     parameters.grid_refinement_study_param = this->all_parameters->grid_refinement_study_param;
+     parameters.artificial_dissipation_param = this->all_parameters->artificial_dissipation_param;
+     parameters.flow_solver_param = this->all_parameters->flow_solver_param;
+     parameters.mesh_adaptation_param = this->all_parameters->mesh_adaptation_param;
+     parameters.artificial_dissipation_param = this->all_parameters->artificial_dissipation_param;
+     parameters.dimension = this->all_parameters->dimension;
+     parameters.pde_type = this->all_parameters->pde_type;
+     parameters.test_type = this->all_parameters->test_type;
+     parameters.use_weak_form = this->all_parameters->use_weak_form;
+     parameters.use_collocated_nodes = this->all_parameters->use_collocated_nodes;
+     parameters.use_periodic_bc = this->all_parameters->use_periodic_bc;
+     
+     parameters.ode_solver_param.initial_time_step *= pow(refine_ratio,refinement);
+     //ADD : flow_solver params: change file name
+     return parameters;
+ }
+
+
+/*
+template <int dim, int nstate>
+void TimeRefinementStudyAdvection<dim, nstate>::write_convergence_summary(int refinement, double dt) {
```

Comment:
Should this be commented? Remove if it won't get used

Reply:
Forgot to remove that, it won't be used. Will fix in next commit.